### DISCUSSION TITLE
Let renovate update git-submodules only when files in them change

### DISCUSSION THREAD

**Discussion by buehler** [renovatebot/renovate#14230] [Suggest an Idea]
### What would you like Renovate to be able to do?

I am currently using the git submodules feature in multiple repositories that I maintain (open-source stuff). While it is a breeze to have renovate update the sub modules when they receive new commits, it would be awesome to config the submodules further.

Especially, I'd like to be able to config certain file paths (with glob style), such that renovate only creates an update PR when one of the changed files hits the configured paths. The "severity" can already be configured (AFAIK).

As an example, I maintain a gRPC API library for [ZITADEL](https://github.com/caos/zitadel), named [zitadel-rust](https://github.com/buehler/zitadel-rust). In this repository, I use submodules to fetch the ZITADEL repository since it contains the needed gRPC `*.proto` files. Now, my desired configuration would be:
```json
{
  "git-submodules": {
    "enabled": true
    "packageRules": [
      // Select the ZITADEL submodule
      "changes": ["src/zitadel/proto/**/*.proto"],
      "semanticCommitType": "fix"
    ]
  }
}
```

Other projects that are included should work by the same principle. For example, since gRPC is used, `googleapis` repository is also linked. They change quite a lot of things (like bazel compiler files and stuff) which are not relevant for "my project". Therefore, it would be cool to filter out files that are not relevant for the given project when updating submodules.

### If you have any ideas on how this should be implemented, please tell us here.

For configuration, I'd suggest that one can include/exclude files in the git submodules config object and the package rules. For the submodules, I could imagine that renovate checks the changed files. This could be possible with git diff or analyzing the commits and listing changed files. I guess, this would require that renovate checks-out the repositories in question. This can be similar to what github actions and other ci pipelines support that certain workflows only run when certain files changed.

### Is this a feature you are interested in implementing yourself?

Maybe

**Comment by viceice** [comment]
That would be pretty hard to implement, as we don't have a concept / api in renovate for such a feature.

I also think that this would crate a lot of git operations / api calls. So i don't like it.

**Reply by rarkins** [reply]
I've moved this to an Ideas discussion, as I mostly agree.

In future we could maybe support more flexible preUpdate/postUpdate concepts which would "abort" a branch, but we're not there yet
